### ISSUE TITLE
tmux-start show context in wrong pane

### ISSUE THREAD

**Issue by freingruber** [hugsy/gef#387]
Your issue will be closed unless you confirm the following:
  * [x] Did you use the latest version of GEF from `master` branch?
  * [x] Did you read the [documentation](https://gef.readthedocs.org/en/latest/) first?
  * [x] Did you check [closed issues](https://github.com/hugsy/gef/issues)?

### Step 1: Describe your environment

  * Operating System: Kali Roling 4.18.10-2
  * Architecture: x64
  * GDB version (including the Python library version): GDB 8.1 with python 3.6

### Step 2: Describe your problem
If I use "tmux-setup" the panes are split, however, the context is still printed on the left pane and not on the right pane.

#### Steps to reproduce

1. gdb ./test
2. tmux-start
3. start

#### Observed Results

  * The context is displayed in the left pane and not the right pane. Also if I do a "si" the context is still printed on the left pane and not the right one.

#### Expected results

  * Context should be printed on the other pane.

#### Analysis

*) Problem does not occure if I enter:

1. gdb ./test
2. gef config gef.debug True  # <-------- added command 
3. tmux-start
4. start

Then the context is correctly displayed on the right pane (and not the left one).
Reason: In my opinion the problem is in the function gef_print()

```
def gef_print(x="", *args, **kwargs):
    """Wrapper around print(), using string buffering feature."""
    if __gef_int_stream_buffer__ and not is_debug():
        return __gef_int_stream_buffer__.write(x + kwargs.get("end", "\n"))
    return print(x, *args, **kwargs)
```

The context is printed with gef_print(). The difference is that without debug mode it prints with __gef_int_stream_buffer__.write() which is not working. If I comment these two lines out and just print with the last line (return print(...)), then it works.

**Comment by Grazfather** [comment]
Confirmed. `gdb.write("test", gdb.STDLOG)` doesn't seem to obey the logging settings.

```
gef➤
Future logs will be written to /dev/pts/4.
Logs will overwrite the log file.
Output will be logged and displayed.
gef➤  pi gdb.write("test", gdb.STDLOG)
testgef➤  ni
```

**Comment by Burner0x00Sec** [comment]
Still confirming this issue. This is rather frustrating..

**Comment by hugsy** [comment]
@Burner0x00Sec Has anybody reported it to GDB devs ?

**Comment by yakamoz147** [comment]
I am also experiencing the same issue (with tmux-setup but also generally with redirecting the context from one tty to another) but I noticed that if I turn debug mode on (gef config gef.debug 1) than the redirect works as expected. Does that still mean its a GDB issue?

**Comment by amlamarra** [comment]
Just wanted to say that I'm having this same issue. I'm on Windows 10 using WSL via Cmder.

**Comment by phra** [comment]
`tmux-setup` is broken too, the context is printed in the left pane while the right pane remains empty.

**Comment by hugsy** [comment]
As said several times in the thread, this is a GDB bug. There is nothing that we, GEF devs can do about it. Please report the bug to GDB's bugzilla.

**Comment by hugsy** [comment]
This should be fixed by commit 3a0203d (in `dev` branch only for now).

Please test and let us know.

Thanks!

**Comment by Grazfather** [comment]
Works for me!
